Added more details for FAQs. (#101)
praveenscience/Internship-LMS-FrontEnd: ReadMe.md

* Salma Shaik
* Prachi Mane

> ## General FAQs

Reviewer: Missing Section heading.

Author: Sir I didn't get that. How to do that.

Reviewer: I'll fix it...

Author: Okay Sir. Sir now can you merge my Pull rerquest. If it seems okay to you.

> ## General FAQs

Reviewer: ```suggestion
## FAQs

### General FAQs
```

> ## General FAQs.

Reviewer: ```suggestion
## FAQs

### General FAQs
```

* Salma Shaik
* Prachi Mane

## General FAQs.

#### What is OpenSource Projects

>  OpenSource Projects means anybody is free to use, study and modify your project. The open source community provides a great opportunity for aspiring programmers to distinguish themselves; and by contributing to various projects, developers can improve their skills and get inspiration and support from like-minded people. But most importantly, they can prove that they can build fantastic experiences that people love. 

Reviewer: - Project, not projects.
- an Open Source.
- remove the space before the first line.
- remove the space after the para.

> ### Why is the master so important to not mess with?

Reviewer: change master to `master` branch. (add back ticks and branch)

> The master branch is deployable. It is your production code, ready to roll out into the world. The master branch is meant to be stable, and it is the social contract of open source software to never, ever push anything to master that is not tested. If you make changes to the master branch while other people are also working on it, your on-the-fly changes will ripple out to affect everyone else and very quickly there will be merge conflicts. So,branching is a technique that gives a developer or group of developers the opportunity to create a new copy of a project without affecting the original which is the master branch. This allows developers to experiment with a branch and if the experiment is successful, Git makes it easy to incorporate the experimental elements into the Master.

Reviewer: - space after comma.
- comma after "group of developers"
- comma before which.

## FAQs

### General FAQs

> #### What is OpenSource Project

Reviewer: ```suggestion
#### What is an Open Source Project?
```

> OpenSource Project means anybody is free to use, study and modify your Project. An Open source community provides a great opportunity for aspiring programmers to distinguish themselves; and by contributing to various Project, developers can improve their skills and get inspiration and support from like-minded people. But most importantly, they can prove that they can build fantastic experiences that people love.

Reviewer: ```suggestion
Open Source Project denotes the project's source code is free for anyone to use, study and modify. An Open Source community provides a great opportunity for aspiring programmers to distinguish themselves; and by contributing to various projects, developers can improve their skills and get inspiration and support from like-minded people. But most importantly, they can prove that they can build fantastic experiences that people love.
```

### Why is the `master` branch so important to not mess with?

> The master branch is deployable. It is your production code, ready to roll out into the world. The master branch is meant to be stable, and it is the social contract of open source software to never, ever push anything to master that is not tested. If you make changes to the master branch while other people are also working on it, your on-the-fly changes will ripple out to affect everyone else and very quickly there will be merge conflicts. So, branching is a technique that gives a developer, group of developers the opportunity to create a new copy of a project without affecting the original, which is the master branch. This allows developers to experiment with a branch and if the experiment is successful, Git makes it easy to incorporate the experimental elements into the Master.

Reviewer: ```suggestion
The master branch is deployable. It is your production code, ready to roll out into the world. The master branch is meant to be stable, and it is the social contract of open source software to never, ever push anything to master that is not tested. If you make changes to the master branch while other people are also working on it, your on-the-fly changes will ripple out to affect everyone else and very quickly there will be merge conflicts. So, branching is a technique that gives a developer, group of developers the opportunity to create a new copy of a project without affecting the original, which is the master branch. This allows developers to experiment with a branch and if the experiment is successful, Git makes it easy to incorporate the experimental elements into the Master.

```

### Why is the `master` branch so important to not mess with?

The master branch is deployable. It is your production code, ready to roll out into the world. The master branch is meant to be stable, and it is the social contract of open source software to never, ever push anything to master that is not tested. If you make changes to the master branch while other people are also working on it, your on-the-fly changes will ripple out to affect everyone else and very quickly there will be merge conflicts. So, branching is a technique that gives a developer, group of developers the opportunity to create a new copy of a project without affecting the original, which is the master branch. This allows developers to experiment with a branch and if the experiment is successful, Git makes it easy to incorporate the experimental elements into the Master.

> 
> 

Reviewer: ```suggestion

```

## FAQs

> #### About OpenSource Projects

Reviewer: Make it in a question way, and also, put this as a Level 3 heading. If not, add another Level 3 heading above it, saying General FAQs.

Author: Okay sir

>  OpenSource Projects means anybody is free to use, study and modify your project for any purpose. The open source community provides a great opportunity for aspiring programmers to distinguish themselves; and by contributing to various projects, developers can improve their skills and get inspiration and support from like-minded people. But most importantly, they can prove that they can build fantastic experiences that people love. means anybody is free to use, study, modify, and distribute your project for any purpose.

Reviewer: Looks good... Please do a grammar check and make it right.

Author: Okay sir

### How to start Contributing in OpenSource Projects

For doing this we must have to go step by step:-
* Fork the repository.
* Clone Your Repository.
* Copy the link.
* Go to git bash and write a command $ git clone https://<User Name>@link.
* Go to your current branch.
* $ git checkout -b <issue Your are working on>       (This will change the branch from master branch)
* Make the changes required.
* After making changes save it and again go to git bash
* $ git status    (This will show which files You have modified)
* $ git diff      (This will show you the exact changes in green color)
* $ git add . (This will add all modifed files)
* $ git commit -m "Fixed ISSUE
   >
   > ISSUE NO."
* $ git push origin <Current branch name>.
* Copy the Pull request link.
* Open the link in your browser.
* Scroll down to see all  changes you have done.

> * Click on option Create Pull Request.

Reviewer: Already Available. You can remove this. 😊

Author: Okay sir.

### Never mess with your master branch

> The master branch is deployable. It is your production code. The master branch is meant to be stable, and it is the social contract of open source software to never, ever push anything to master that is not tested, or that breaks the build.  If you make changes to the master branch of a group project while other people are also working on it, your on-the-fly changes will ripple out to affect everyone else and very quickly there will be merge conflicts.

Reviewer: Super cool. 😊  Again, grammar changes please?

Author: Sure sir

#### Can I work on an already opened issue?

> You can always ask in the comments section of the opened issue to get yourself assigned but the owner may or may not assign you that issue. However, the owner may suggest you a fresh issue (means by creating new issue).

Reviewer: ```suggestion
You can always ask in the comments section of the opened issue to get yourself assigned but the owner may or may not assign you that issue. However, the owner may suggest you a fresh issue (either by creating a new issue or requesting you to create a new issue).
```

#### Can I work on an already opened issue?

> You can always ask in the comments section of the opened issue to get yourself assigned but the owner may or may not assign you that issue. However, the owner may suggest you a fresh issue (in which you have to create new issue by reading about that Project,searchig the bug present in it).

Reviewer: Grammar changes, spelling mistakes to be corrected please. 😊

Author: sure sir.
